File: src/main/java/textGame/HealingItemFactory.java
package textGame;

public class HealingItemFactory extends GameObjectCsvFactory {

    @Override
    public Item create(String[] itemInformation) {
        HealingItem healingItem = new HealingItem();

        healingItem.setName(itemInformation[0]);
        healingItem.setDescription(itemInformation[1]);
        healingItem.setHealAmount(Integer.parseInt(itemInformation[2]));
        healingItem.setUses(Integer.parseInt(itemInformation[3]));

        return healingItem;
    }

    @Override
    public String getCsvPath() {
        return "src/main/resources/healing_items.csv";
    }
}

File: src/main/java/textGame/HealingItemFactory.kt
package textGame

import kotlin.jvm.internal.ClassReference

class HealingItemFactory : GameObjectFactory() {
    override fun create(gameObjectInformation: Array<String>): GameObject {
        val healingItem = HealingItem()

        healingItem.setName(gameObjectInformation[0])
        healingItem.setDescription(gameObjectInformation[1])
        healingItem.healAmount = gameObjectInformation[2].toInt()
        healingItem.setUses(gameObjectInformation[3].toInt())

        return healingItem
    }

    override fun getCsvPath(): String {
        return "src/main/resources/healing_items.csv"
    }
}